src/models/event_query.cr

```diff
@@ -7,10 +7,15 @@ class Office365::EventQuery
   end
 
   def self.from_json(data)
-    parsed_data = JSON.parse(data).as_h["value"].as_a
-    result = parsed_data.map do |event|
-      Event.from_json(event.to_json)
+    parsed_data = JSON.parse(data).as_h
+    if parsed_data.has_key?("value")
+      array_data = parsed_data["value"].as_a
+      result = array_data.map do |event|
+        Event.from_json(event.to_json)
+      end
+      new(value: result)
+    else
+      raise JSON::Error.new "Query not in the correct query format\nOffice365 Error: #{parsed_data["error"]["message"]?}"
```

Comment:
Strict access here could result in an `KeyError`. Given there's potentially something wild leading to this (as the error is not being caught [here](https://github.com/PlaceOS/office365/blob/7f0f8235c20a959be7c541007da16c58e2818d8e/src/client.cr#L73-L77)) it would be better to try and extract that message (if there's one there), or default to a non-parsed version.